### PR TITLE
ENH: add basic socket.socket/synchronous repeater

### PR DESCRIPTION
**Pull request by klauer**
Switches default repeater to the synchronous repeater

- [x] TODO: rebase and remove unrelated commit
- [x] Coverage of sync repeater
- [x] Tests passing about as well as the asyncio repeater.

### CODE REVIEW

**Comment by klauer** [comment]
TODO: launch repeater with coverage

**Comment by tacaswell** [comment]
Bit concerned about how much of the diff is missed by tests.  Are the async tests failing to actually run?

**Comment by klauer** [comment]
Agreed, @tacaswell - anyone want to help look into the coverage issues? One of the repeaters has to be running, and yet neither show adequate coverage on travis.

File: caproto/sync/repeater.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    return sock
+
+
+def run(host='0.0.0.0'):
+    port = caproto.get_environment_variables()['EPICS_CA_REPEATER_PORT']
+    addr = (host, port)
+    logger.debug('Checking for another repeater')
+
+    # NOTE: This check is performed separately because
+    # loop.create_datagram_endpoint is not consistent on different platforms
```
[the remark this answers is not in the document]
Comment (klauer) [reply caproto/sync/repeater.py]: Fixed

**Comment by klauer** [comment]
I think this is ready. Actual coverage on sync repeater is about 89%, despite the above codecov comment. Asyncio (non-default) is around 33%, but I haven't investigated that.

**Pull request merged**